### PR TITLE
CI: Add `concurrency` to CI-Mac/Win + newer-Java workflows

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: .github/workflows/ci-mac.yml [hunk 1]
```diff
@@ -25,6 +25,16 @@ on:
...
   workflow_dispatch:
 
+# For the main branch: let all CI runs complete, one after the other. This has a couple advantages:
+# * Site deployments happen in commit-order
+# * Saved Gradle cache are persisted in commit-order
+# * (Potentially) more GH runners available for PRs
+concurrency:
+  # PRs: 1 CI run concurrently / older ones are cancelled
+  # main branch: 1 CI run concurrently / all commits / no cancellation
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.ref }}
```
Comment (XN137) [inline .github/workflows/ci-mac.yml:35]: just wondering: why is it important to include event_name ?

**Review by XN137 — APPROVED** [review]
concurrency changes LGTM, not sure i understand why the java 17 ones are included

File: .github/workflows/ci-mac.yml [hunk 2]
```diff
@@ -43,7 +53,7 @@ jobs:
       uses: gradle/gradle-build-action@v2
       env:
         # Same as for ci.yml
-        GRADLE_BUILD_ACTION_CACHE_KEY_ENVIRONMENT: java-11
+        GRADLE_BUILD_ACTION_CACHE_KEY_ENVIRONMENT: java-17
```
[the remark this answers is not in the document]
Comment (snazy) [reply .github/workflows/ci-mac.yml:56]: oops - that should be in #7783

File: .github/workflows/newer-java.yml [hunk 3]
```diff
@@ -17,7 +27,6 @@ jobs:
       max-parallel: 1
       matrix:
         include:
-          - java-version: 17
```
Comment (XN137) [inline .github/workflows/newer-java.yml:20]: one more ?!

Comment (snazy) [reply .github/workflows/newer-java.yml:20]: Let's leave that - it'll go away w/ #7783 anyway

**Pull request merged**